url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/net/http/SimpleHttpServer.java

```diff
@@ -0,0 +1,113 @@
+package net.hetimatan.net.http;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+import net.hetimatan.io.file.KyoroFile;
+import net.hetimatan.io.file.KyoroFileForFiles;
+import net.hetimatan.io.filen.ByteKyoroFile;
+import net.hetimatan.io.filen.CashKyoroFileHelper;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.net.http.HttpServerFront;
+import net.hetimatan.util.event.net.io.KyoroSocket;
+import net.hetimatan.util.http.HttpObjectHelper;
+import net.hetimatan.util.http.HttpRequest;
+
+//@todo
+public class SimpleHttpServer extends HttpServerTemplate {
+
+	private CashKyoroFile mFile = null;
+	private LinkedHashMap<String, String> mMimetype = new LinkedHashMap<String,String>();
+	{
+		mMimetype.put("mp4", "video/mp4");
+		mMimetype.put("txt", "text/plain");
+		mMimetype.put("html", "text/html");
+	}
+
+	public SimpleHttpServer() { 
+		try {
+			mFile = new CashKyoroFile(new File("../../h264.mp4"), 16*1024, 4);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+
+	@Override
+	public KyoroFile createResponse(HttpServerFront front, KyoroSocket socket, HttpRequest uri) throws IOException {
+		String rangeHeader = uri.getHeaderValue("Range");
+		boolean isRange = false;
+		PieceInfoList list = null;
+		if(rangeHeader != null && rangeHeader.length() != 0) {
+			list = HttpObjectHelper.getRangeList(rangeHeader,mFile.length());
+			if(list.size()>0) {
+				isRange = true;
+			}
+		}
+		if(!isRange) {
+			return createDefaultResponse(front, socket, uri);
+		} else {
+			return createSingleRangeResponse(list.getPieceInfo(0), front, socket, uri);
+//			return createMultiRangeResponse(list.getPieceInfo(0), front, socket, uri);
+		}
+	}
+
+	public String getMimeType(String path) {
+		Set<String> keyset = mMimetype.keySet();
+		for(String key:keyset) {
+			if(path.endsWith(key)){
+				return mMimetype.get(key);
+			}
+		}
+		return "text/plain";
+	}
+
+	public KyoroFile createDefaultResponse(HttpServerFront front, KyoroSocket socket, HttpRequest uri) throws IOException {
+		KyoroFile content = CashKyoroFileHelper.subSequence(mFile, 0, mFile.length());
+		KyoroFile header = new ByteKyoroFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *header*. Currently, there are execution paths that do not contain closure statements, for example, when KyoroFile.addChunk throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).